packages/shared_preferences/shared_preferences_android/lib/shared_preferences_android.dart

```diff
@@ -7,47 +7,63 @@ import 'dart:async';
 import 'package:flutter/services.dart';
 import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
 
-const MethodChannel _kChannel =
-    MethodChannel('plugins.flutter.io/shared_preferences_android');
+import 'messages.g.dart';
 
 /// The Android implementation of [SharedPreferencesStorePlatform].
 ///
 /// This class implements the `package:shared_preferences` functionality for Android.
 class SharedPreferencesAndroid extends SharedPreferencesStorePlatform {
+  final SharedPreferencesApi _api = SharedPreferencesApi();
+
   /// Registers this class as the default instance of [SharedPreferencesStorePlatform].
   static void registerWith() {
     SharedPreferencesStorePlatform.instance = SharedPreferencesAndroid();
   }
 
   @override
   Future<bool> remove(String key) async {
-    return (await _kChannel.invokeMethod<bool>(
-      'remove',
-      <String, dynamic>{'key': key},
-    ))!;
+    return _api.remove(key);
   }
 
   @override
   Future<bool> setValue(String valueType, String key, Object value) async {
-    return (await _kChannel.invokeMethod<bool>(
-      'set$valueType',
-      <String, dynamic>{'key': key, 'value': value},
-    ))!;
+    return await _handleSetValue(valueType, key, value) ?? false;
```

Comment:
Earlier set`DataType` was used, as at the `Android` end we used to get its name from `callMethod`. However, it makes more sense to filter that at the Dart End. Either way, this function has to be created at either Dart or Android end (which maps `setDataType` string to a function according to the dataType).

Reply:
I'm not asking why the code that's currently in the helper exists, I'm asking why it's in a helper that's only called from one place, where that place ends up being just a single line calling that helper, rather than that code being here. I don't see what value the indirection is providing.